plugins/woocommerce/includes/wc-template-functions.php

```diff
@@ -2854,7 +2854,8 @@ function woocommerce_form_field( $key, $args, $value = null ) {
 
 		if ( $args['required'] ) {
 			$args['class'][] = 'validate-required';
-			$required        = '&nbsp;<abbr class="required" title="' . esc_attr__( 'required', 'woocommerce' ) . '">*</abbr>';
+			$required        = '&nbsp;<abbr class="required" title="' . esc_attr__( 'required', 'woocommerce' ) . '" aria-hidden="true">*</abbr><span class="screen-reader-text">' . esc_html__( "Required", "woocommerce" ) . '</span>';
+			$args['custom_attributes']['aria-required'] = 'true';
```

Comment:
Prevent screen readers from announcing the asterisk and make them announce the term "required" instead.